src/lava/proc/lif/models.py

```diff
@@ -97,17 +92,27 @@ def __init__(self):
         self.vth_shift = 6
         self.act_shift = 6
 
-    def run_spk(self):
-        # Receive synaptic input
-        a_in_data = self.a_in.recv()
+    def scale_bias(self):
+        """Scale bias with bias exponent by taking into account sign of the
+        exponent"""
+        self.effective_bias = np.where(self.bias_exp >= 0, np.left_shift(
+            self.bias, self.bias_exp), np.right_shift(self.bias,
+                                                      -self.bias_exp))
+        # self.effective_bias = np.left_shift(self.bias, self.bias_exp) if \
+        #     self.bias_exp >= 0 else np.right_shift(self.bias, -self.bias_exp)
+        self.isbiasscaled = True
 
-        # Compute effective bias and threshold only once, not every time-step
-        if not self.b_vth_computed:
-            self.effective_bias = np.left_shift(self.bias, self.bias_exp)
-            # In Loihi, user specified threshold is just the mantissa, with a
-            # constant exponent of 6
-            self.effective_vth = np.left_shift(self.vth, self.vth_shift)
-            self.b_vth_computed = True
+    def scale_threshold(self):
+        """Placeholder method for scaling threshold(s)."""
+        pass
+
+    def spiking_activation(self):
+        """Placeholder method to specify spiking behaviour of a LIF neuron."""
+        return self.s_out
```

Comment:
Is this a common behavior? If not, it's best to have `pass` here or maybe `raise NotImplementedError` to ensure the derived class always overwrites this method.

Reply:
Same for float version as well

Reply:
I am using this function in run_spk to return spiking activation. If it does not return anything here, I will get PEP related style errors/warnings.. PyBuilder Analyze catches these.

Reply:
I don't think that would generate style warnings. Must be something else.

Reply:
Resolved.